model-engine/model_engine_server/api/app.py

```diff
@@ -32,12 +33,29 @@
 
 logger = make_logger(logger_name())
 
+# TODO: determine this based on load testing; for now, setting to 4x the per-pod targetConcurrency of 25
+MAX_CONCURRENCY = 100
+
+concurrency_limiter = MultiprocessingConcurrencyLimiter(
+    concurrency=MAX_CONCURRENCY, fail_on_concurrency_limit=True
+)
+
 
 class CustomMiddleware(BaseHTTPMiddleware):
     async def dispatch(self, request: Request, call_next):
         try:
-            LoggerTagManager.set(LoggerTagKey.REQUEST_ID, str(uuid.uuid4()))
-            return await call_next(request)
+            with concurrency_limiter:
```

Comment:
What do people think about trying this out with just a specific route at first? Looking at the breakdown in the [past week](https://app.datadoghq.com/metric/explorer?start=1701146189478&end=1701750989478&paused=false#N4Ig7glgJg5gpgFxALlAGwIYE8D2BXJVEADxQEYAaELcqyKBAC1pEbghkcLIF8qo4AMwgA7CAgg4RKUAiwAHOChASAtnADOcAE4RNIKtrgBHPJoQaUAbVBGN8qVoD6gnNtUZCKiOq279VK7ueJiWyDYgGGgQGGEgACo4CFEABACiAG5wIhYGIEGqIRjKpjpYZCA8ALpUpf5hEVCeGE4a+NoAxkrIIOoIuh2WVCIY6iVm2uV5dTQ9GniqyBAaEjgAdL6Ma0alK2sd+DlrCElRwB1oeCs6TiPqyKpoALR+GaIwTyJwYCkUKQIrUSeSQiJwYeTyB44ATPbIwURwP4AQh2ZhWThwim0wKkyDwIgA1iIcGARDwUgAjLApYCo8wYrE4slrWJOA74hAACgAlJUaiAVlg0N1QPIoogEN0QFAcDA2aENBAOnlol8nHJFMo2tEoCqEU56ExlMT3FFKvxlvJMFh1QopRSMNpLNUeHwBVbxABhKTCGAoEQhNA8IA), `get_/v1/async-tasks/_task_id` is the most common route by far. 

I'm not sure if it make sense to do a global limit, as we know some routes take more time than others.